sklearn/semi_supervised/_self_training.py

```diff
@@ -237,6 +284,12 @@ def fit(self, X, y):
                 UserWarning,
             )
 
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **params)
+        else:
+            routed_params = Bunch()
+            routed_params.estimator = Bunch(fit=params)
```

Comment:
Hasn't params been added in this PR? In that case this is the default without routing and probably should be empty?
```suggestion
            routed_params.estimator = Bunch(fit={})
```

Reply:
Right! Let me fix that… I think I misunderstood the comment from https://github.com/scikit-learn/scikit-learn/pull/28494#discussion_r1647370405